lib/constructs/DynamoDBToolboxPutItem.unit.test.ts

```diff
@@ -0,0 +1,303 @@
+import { App, Stack, DefaultStackSynthesizer, StackProps } from "aws-cdk-lib";
+import type { Construct } from "constructs";
+import {
+  StateMachine,
+  StateMachineType,
+  Succeed,
+} from "aws-cdk-lib/aws-stepfunctions";
+
+import { DynamodbToolboxPutItem } from "./DynamodbToolboxPutItem";
+import { TestEntity } from "./DynamoDBToolboxPutItem.fixtures.test";
+
+const app = new App();
+
+class TestStack extends Stack {
+  constructor(scope: Construct, constructId?: string, props?: StackProps) {
+    super(scope, constructId, props);
+
+    const { chain } = new DynamodbToolboxPutItem(
+      this,
+      "DynamodbToolboxPutItem",
+      {
+        // @ts-ignore this is not resolved yet
+        entity: TestEntity,
+      }
+    );
+
+    new StateMachine(this, "StepFunction", {
+      stateMachineName: "SaveAnimalStepFunction2",
+      definition: chain.next(new Succeed(scope, "SuccessTask")),
+      stateMachineType: StateMachineType.EXPRESS,
+    });
+  }
+}
+
+const testStack = new TestStack(app, "testStack", {
+  // Override default synthetizer to prevent check related to CDK bootstrap
+  synthesizer: new DefaultStackSynthesizer({
+    generateBootstrapVersionRule: false,
+  }),
+});
+
+type Resource = { Type: string; Properties?: { [key: string]: any } };
+
+const stackTemplate = app.synth().getStackByName(testStack.stackName)
+  .template as { Resources: Record<string, Resource> };
+
+const stateMachines = Object.values(stackTemplate.Resources).filter(
+  (resource) => resource.Type === "AWS::StepFunctions::StateMachine"
+);
+
+describe("test", () => {
```

Comment:
describe("template", () => {
  it("should provision exactly one state machine", ...